thorlcr/graph/thgraphslave.cpp

```diff
@@ -140,10 +140,29 @@ void CSlaveActivity::setInput(unsigned index, CActivityBase *inputActivity, unsi
     else
         outLink.set(((CSlaveActivity *)inputActivity)->queryOutput(inputOutIdx));
     assertex(outLink);
+
     while (inputs.ordinality()<=index) inputs.append(NULL);
+
     inputs.replace(outLink.getClear(), index);
 }
 
+void CSlaveActivity::appendOutput(IThorDataLink *itdl)
+{
+    if (queryJob().getOptBool("TRACEROWS"))
+    {
+        const unsigned numTraceRows = queryJob().getOptInt("numTraceRows", 10);
+        outputs.append(new CTracingThorDataLink(itdl, queryHelper(), numTraceRows));
```

Comment:
No, if outputs is cast back to the original base class, it won't work.  We had to make changes to NSplitterSlaveActivity because of this issue.  Let me see everywhere that outputs and, if necessary, modify to work with CTracingThorDataLink.